tests/app/main/views/test_users.py

```diff
@@ -271,39 +282,51 @@ def test_get_form_with_valid_framework(self, data_api_client):
         assert response.status_code == 200
         self._assert_things_about_frameworks(response, frameworks)
 
+    @pytest.mark.parametrize("role, expected_code", [
+        ("admin", 200),
+        ("admin-ccs-category", 403),
+        ("admin-ccs-sourcing", 403),
+        ("admin-framework-manager", 200),
+        ("admin-manager", 403),
+    ])
+    def test_download_users_page_only_accessible_to_specific_user_roles(self, data_api_client, role, expected_code):
+        self.user_role = role
+        frameworks = [self._valid_framework]
+        response = self._return_get_user_export_response(data_api_client, frameworks)
+        actual_code = response.status_code
+        assert actual_code == expected_code, "Unexpected response {} for role {}".format(actual_code, role)
+
     def test_user_export_with_one_user(self, data_api_client):
         framework = self._valid_framework
-        users = [{
-            "application_result": "fail",
-            "application_status": "no_application",
-            "declaration_status": "unstarted",
-            "framework_agreement": False,
-            "supplier_id": 1,
-            "email address": "test.user@sme.com",
-            "user_name": "Test User",
-            "variations_agreed": "var1"
-        }]
+        users = [self._supplier_user]
 
         response = self._return_user_export_response(data_api_client, framework, users)
         assert response.status_code == 200
 
     def test_download_csv(self, data_api_client):
         framework = self._valid_framework
-        users = [{
-            "application_result": "fail",
-            "application_status": "no_application",
-            "declaration_status": "unstarted",
-            "framework_agreement": False,
-            "supplier_id": 1,
-            "email address": "test.user@sme.com",
-            "user_name": "Test User",
-            "variations_agreed": "var1"
-        }]
+        users = [self._supplier_user]
 
         response = self._return_user_export_response(data_api_client, framework, users)
         assert response.status_code == 200
         self._assert_things_about_user_export(response, users)
 
+    @pytest.mark.parametrize("role, expected_code", [
+        ("admin", 200),
+        ("admin-ccs-category", 403),
+        ("admin-ccs-sourcing", 403),
+        ("admin-framework-manager", 200),
+        ("admin-manager", 403),
+    ])
+    def test_supplier_csv_is_only_accessible_to_specific_user_roles(self, data_api_client, role, expected_code):
```

Comment:
Weird how `test_download_user_lists_link_is_shown_to_users_with_the_right_role` checks that Support don't see a link to the lists, but then we test that `admin` can download the lists if they visit the URL directly.  I think it's OK for here, as I'm working on splitting the lists and will remove `admin` from being able to do so.